pages/experiments/[id].tsx

```diff
@@ -0,0 +1,37 @@
+import debugFactory from 'debug'
+import { useRouter } from 'next/router'
+import React, { useEffect, useState } from 'react'
+import Container from 'semantic-ui-react/dist/commonjs/elements/Container'
+
+import AnalysesApi from '@/api/AnalysesApi'
+
+import ErrorsBox from '@/components/ErrorsBox'
+import Layout from '@/components/Layout'
+
+import { Analysis } from '@/models'
+
+const debug = debugFactory('abacus:pages/index.tsx')
+
+export default function ExperimentPage() {
+  const experimentId = Number(useRouter().query.id)
```

Comment:
I wouldn't cast to a number here, it hides a few things. `Number(null) === 0`, `Number('hi') === NaN`, `Number(012) === 10`, instead do some sanity checks and cast in your effect where you can set an error there.

```suggestion
```

Reply:
Can use the `qc-to_int` package. It doesn't have the issues `parseInt` or `Number` have. If it can't be parsed and an integer, then the default is returned. This is so that NaN, which in most cases isn't helpful, isn't returned unless that is what you want the default to be.

```
import { toInt } from 'qc-to_int'
```

```suggestion
  const experimentId = toInt(useRouter().query.id, null)
```

It also exports a `toIntOrNull` function.